nerfstudio/model_components/losses.py

```diff
@@ -502,3 +502,29 @@ def tv_loss(grids: TensorType["grids", "feature_dim", "row", "column"]) -> Tenso
     h_tv = torch.pow((grids[:, :, 1:, :] - grids[:, :, :-1, :]), 2).sum()
     w_tv = torch.pow((grids[:, :, :, 1:] - grids[:, :, :, :-1]), 2).sum()
     return 2 * (h_tv / h_tv_count + w_tv / w_tv_count) / number_of_grids
+
+
+class GradientScaler(torch.autograd.Function):  # typing: ignore, pylint: disable=abstract-method
+    """
+    Scale gradients by the ray distance to the pixel
+    as suggested in `Radiance Field Gradient Scaling for Unbiased Near-Camera Training` paper
+
+    Note, the scaling is applied on the interval of [0, 1] along the ray!
+    """
+
+    @staticmethod
+    def forward(ctx, field_outputs, ray_samples):  # pylint: disable=arguments-differ
```

Comment:
Can we type and add dosctrings? Or does it break since we are overwriting?

Reply:
Typing and doctrings are not visible anywhere. I have added a wrapper function to provide the docstrings and typing support.